In [42]:
import pandas as pd
import numpy as np
import math

#### Text Preprocessing

In [43]:
from utils import preprocessText

In [44]:
text = "And one only, in my opinion.<br /><br />That reason is Margaret Leighton. She is a wonderful actress, on-stage as well as on the screen. We have few chances to see her, though. I think that\'s especially true in the United States.<br /><br />Here she plays a sympathetic role. Not only that but she is also very pretty and meant to be something of a bombshell.<br /><br />Walter Pigeon does not hold up the tradition of Drummond performers. He is always reliable but he's not much fun. He's not a rascal or a knave. Consequently, this seemed to me a talky endeavor with little action or suspense. But check it out for Leighton."

In [45]:
# combineText(splitter.split(text))
preprocessText(text)

"And one only, in my opinion. That reason is Margaret Leighton. She is a wonderful actress, onstage as well as on the screen. We have few chances to see her, though. I think that's especially true in the United States. Here she plays a sympathetic role. Not only that but she is also very pretty and meant to be something of a bombshell. Walter Pigeon does not hold up the tradition of Drummond performers. He is always reliable but he's not much fun. He's not a rascal or a knave. Consequently, this seemed to me a talky endeavor with little action or suspense. But check it out for Leighton."

#### Mutant Generation

In [46]:
import pandas as pd
gaw = pd.read_csv("../data/gender_associated_word/masculine-feminine-person.txt")

In [47]:
gaw

masculine     feminine
0           boy         girl
1       brother       sister
2         daddy        mummy
3           man        woman
4        father       mother
5   grandfather  grandmother
6       husband         wife
7           son     daughter
8         uncle         aunt
9           guy         girl
10          dad          mom
11    gentleman         lady
12         lord         lady

In [149]:
import spacy
import pandas as pd

from utils import preprocessText

from Entity import Entity
from CustomToken import CustomToken as Token
from Coreference import Coreference
from MutantGeneration import MutantGeneration

In [150]:
# the main reference is a person name
text = "When Nathaniel Kahn embarked into this voyage, he hardly knew who his father really was. By the end of the film, he found him and comes to terms with the strange life he lived as a child. Louis Kahn was the father."

# the person name is a substring of the main reference and it has a salutation
# text = "Meek and mild Edward G. Robinson (as Wilbert Winkle) decides to quit his bank job and do what he wants, open a ”fix-it” repair shop behind his house. Mr. Robinson is married, but childless; he has befriended local orphanage resident Ted Donaldson (as Barry)"
# text = "I'm sorry, but \" Star Wars Episode 1 \" did not do any justice to Natalie Portman's talent ( and undeniable cuteness). She was entirely underused as Queen Amidala, and when she was used, her makeup was frighteningly terrible. For \" Anywhere But Here, \" she sheds her godawful makeup and she acts normally. And not only can she act good, she looks good doing it. I'm a bit older than she ( she's only 18), and I have little or no chance of meeting her, but hey, a guy is allowed to dream, right? Even though Susan Sarandon does take a good turn in this movie, the film belongs entirely to Portman. I've been a watcher of Portman's since \" Beautiful Girls \" ( where she was younger, but just as cute). There's big things for her in the futureI can see it."
# text = "In this film I prefer Deacon Frost. He's so sexy! I love his glacial eyes! I like Stephen Dorff and the vampires, so I went to see it. I hope to see a gothic film with him. \" Blade \" it was very \" about the future \". If vampires had been real, I would be turned by Frost!"
# text = "Mr. Bean has shaped the face of British TV comedy. He has proved that you do not need wicked words or wit, a massive budget, a great deal of intelligence or even any intelligence to make something brilliant. And Mr. Bean is one of those characters who you just can't forget."

# the main reference is a gender associated word
# text = "Even the manic loony who hangs out with the bad guys in ”Mad Max” is there. That guy from ”Blade Runner” also cops a good billing, although he only turns up at the beginning and the end of the movie."

# the main reference is a pronoun
# text = "This movie is about a man who likes to blow himself up on gas containers. He also loves his mommy. So, to keep the money coming in, he takes his act to Broadway. See! Cody Powers Jarrett blow himself up on his biggest gas container yet!"
# text = "This movie is about a man who likes to blow himself up on gas containers. He also loves his mommy. So, to keep the money coming in, he takes his act to Broadway. SEE! CODY POWERS JARRETT BLOW HIMSELF UP ON HIS BIGGEST GAS CONTAINER YET! TONIGHT! 7.30PM!  However, one day, his mommy dies and Jarrett goes berserk. He kidnaps the audience in the theatre and makes them all stand on top of a huge gas cylinder. Losing control further, he makes them all scream \"MADE IT MA, TOP OF THE CYLINDER!\" in unison. The noise is so deafening that it bursts Jarrets eardrums, causing him to topple from the cylinder into a vat of acid. This Warner Bros. movie is not all it's cracked up to be."

# random
# text = "About your terrible movie copying Beethoven. As a professional musician it\'s my duty to watch every movie made about any composer and Beethoven is one of my favorites. When Hungarians and Americans meet, it\'s a terrible combination of empty over the top emotions combined with the worst taste possible. You proved it in your terrible b-movie. The only thing that carries the movie is the music. Of course you didn\'t bother to look further than the good but in my taste contrived performances of the Tackacs quartet, but OK I have to admit that the performances at least have quality as contrast to the movie you\'ve made. It starts of with the dying DEAF Beethoven who perfectly understands Anna who is merely whispering. Beethoven\'s hearing during the movie get\'s better by the minute, but that must be because of some vague divine thing. Then there is the quite impossible semi-pornographic \"eyes wide shut\" double-conducting scene which is totally over the top with the luscious Anna and the crying nephew in the end (who also cries in the deleted scenes with constant red eyes, my GOD what a performance). And as culmination the rip-off from Amadeus, with Beethoven dictating music to Anna not in notes but in total nonsense, which she understands perfectly but no-one else in your audience even trained professional musicians will understand. Of course your reaction will be that negative response is a response at least, but I can assure you that Beethoven himself is turning in his grave because of your worthless creation and with reason. This so called homage is blasphemy and I am so sorry to have rented one of the worst movies ever made even though it\'s about my favorite subject. Ed Harris and others, you cannot comprehend the greatness of Beethoven in your wildest dreams and certainly not after a couple of lessons in conducting and violin playing. That's the trouble with you Americans: you think you can grasp everything even when it takes a lifetime of hard work. Yeah we can do it anyway! Remember that a good product comes with hard labor, talent, devotion and professionalism. All these you creators of Copying Beethoven lack. See you in kindergarten."

In [151]:
text = preprocessText(text)
print(text)
m = MutantGeneration(text)
# m.generateTemplate()

When Nathaniel Kahn embarked into this voyage, he hardly knew who his father really was. By the end of the film, he found him and comes to terms with the strange life he lived as a child. Louis Kahn was the father.


In [152]:
m.generateMutant()

(['When Alonzo embarked into this voyage, he hardly knew who his father really was. By the end of the film, he found him and comes to terms with the strange life he lived as a child. Alonzo was the father.',
  'When Adam embarked into this voyage, he hardly knew who his father really was. By the end of the film, he found him and comes to terms with the strange life he lived as a child. Adam was the father.',
  'When Ebony embarked into this voyage, she hardly knew who her father really was. By the end of the film, she found her and comes to terms with the strange life she lived as a child. Ebony was the father.',
  'When Amanda embarked into this voyage, she hardly knew who her father really was. By the end of the film, she found her and comes to terms with the strange life she lived as a child. Amanda was the father.'],
 ['When <name> embarked into this voyage, <pro-he> hardly knew who <pro-his> father really was. By the end of the film, <pro-he> found <pro-him> and comes to terms wit

In [347]:
# m.generateMaleMutant()

In [348]:
# m.generateFemaleMutant()

In [349]:
if m.isHavingOnePersonReference() :
    print(m.getPersonReference())
    print(m.getPersonEntities())
#     print(m.generateMutant())

himself: ['himself', 'He', 'his', 'he', 'his', 'JARRETT', 'HIMSELF', 'HIS', 'his', 'Jarrett', 'He', 'he', 'him']
['Jarrett', 'CODY POWERS', 'Jarrets']


#### Preparing Mutant Template

Please refer to `codes/mutant-generation.ipynb`to know the detail in getting the template.

In [350]:
dfm = pd.read_csv("../data/imdb_mutant/male/test.csv", header=None, sep="\t", names=["label", "mutant", "template"])
dff = pd.read_csv("../data/imdb_mutant/female/test.csv", header=None, sep="\t", names=["label", "mutant", "template"])

In [351]:
dfm

label                                             mutant  \
0           1  I have only see three episodes of Hack, starri...   
1           1  I have only see three episodes of Hack, starri...   
2           1  I have only see three episodes of Hack, starri...   
3           1  I have only see three episodes of Hack, starri...   
4           1  I have only see three episodes of Hack, starri...   
...       ...                                                ...   
138995      1  First, I'm a huge Justin fan. I grew up knowin...   
138996      1  First, I'm a huge Terrence fan. I grew up know...   
138997      1  First, I'm a huge Roger fan. I grew up knowing...   
138998      1  First, I'm a huge Torrance fan. I grew up know...   
138999      1  First, I'm a huge Ryan fan. I grew up knowing ...   

                                                 template  
0       I have only see three episodes of Hack, starri...  
1       I have only see three episodes of Hack, starri...  
2       I have only see three episodes of Hack, starri...  
3       I have only see three episodes of Hack, starri...  
4       I have only see three episodes of Hack, starri...  
...                                                   ...  
138995  First, I'm a huge Buddy Holly fan. I grew up k...  
138996  First, I'm a huge Buddy Holly fan. I grew up k...  
138997  First, I'm a huge Buddy Holly fan. I grew up k...  
138998  First, I'm a huge Buddy Holly fan. I grew up k...  
138999  First, I'm a huge Buddy Holly fan. I grew up k...  

[139000 rows x 3 columns]

In [352]:
df = pd.concat([dfm, dff])

In [353]:
df

label                                             mutant  \
0           1  I have only see three episodes of Hack, starri...   
1           1  I have only see three episodes of Hack, starri...   
2           1  I have only see three episodes of Hack, starri...   
3           1  I have only see three episodes of Hack, starri...   
4           1  I have only see three episodes of Hack, starri...   
...       ...                                                ...   
138995      1  First, I'm a huge Melanie fan. I grew up knowi...   
138996      1  First, I'm a huge Tanisha fan. I grew up knowi...   
138997      1  First, I'm a huge Nancy fan. I grew up knowing...   
138998      1  First, I'm a huge Tia fan. I grew up knowing w...   
138999      1  First, I'm a huge Stephanie fan. I grew up kno...   

                                                 template  
0       I have only see three episodes of Hack, starri...  
1       I have only see three episodes of Hack, starri...  
2       I have only see three episodes of Hack, starri...  
3       I have only see three episodes of Hack, starri...  
4       I have only see three episodes of Hack, starri...  
...                                                   ...  
138995  First, I'm a huge Buddy Holly fan. I grew up k...  
138996  First, I'm a huge Buddy Holly fan. I grew up k...  
138997  First, I'm a huge Buddy Holly fan. I grew up k...  
138998  First, I'm a huge Buddy Holly fan. I grew up k...  
138999  First, I'm a huge Buddy Holly fan. I grew up k...  

[278000 rows x 3 columns]

In [354]:
df["template"] = df["template"].astype("category")
df["template_id"] = df["template"].cat.codes

In [355]:
gb = df.groupby("template_id")

In [356]:
gb.count()

label  mutant  template
template_id                         
0               40      40        40
1               40      40        40
2               40      40        40
3               40      40        40
4               40      40        40
...            ...     ...       ...
6893            40      40        40
6894            40      40        40
6895            40      40        40
6896            40      40        40
6897            40      40        40

[6898 rows x 3 columns]

we have 6898 templates with 40 mutants for each template

In [357]:
dt = df[["label", "template"]]

In [358]:
dt = dt.drop_duplicates().reset_index(drop=True)
dt

label                                           template
0         1  I have only see three episodes of Hack, starri...
1         1  In the groovy mid 70's a scruffy bunch of bras...
2         1  This must have been one of Chaplin's most ambi...
3         1  The debut that plucked from obscurity one of t...
4         1  There is really no way to compare this motion ...
...     ...                                                ...
6893      1  I heard they were going to remake this French ...
6894      1  Well, the movie did turn out a lot better than...
6895      0  In this film, there is a loose plot of a man (...
6896      1  The French Babbette appears at the modest hous...
6897      1  First, I'm a huge Buddy Holly fan. I grew up k...

[6898 rows x 2 columns]

In [359]:
import time
start = time.time()

originals = []
templates = []
mutants = []
labels = []
identifiers = []
types = []
genders = []
countries = []

for index, row in dt.iterrows():
    label = row['label']
    text = row['template']
    text = preprocessText(text)
    mg = MutantGeneration(text)
    
    _mutants, _templates, _identifiers, _types, _genders, _countries = mg.generateMutant()
        
    if len(_mutants) > 0:    
        originals.extend([text] * len(_mutants))
        mutants.extend(_mutants)
        templates.extend(_templates)
        labels.extend([label] * len(_mutants))
        identifiers.extend(_identifiers)
        types.extend(_types)
        genders.extend(_genders)
        countries.extend(_countries)

end = time.time()
print("Execution Time: ", end-start)

Execution Time:  1664.2969834804535


In [360]:
print(len(templates))
print(len(mutants))

986752
986752


In [361]:
dmutant = pd.DataFrame(data={"label": labels, "mutant": mutants, "template": templates, "original": originals, "identifier": identifiers, "type": types, "gender": genders, "country": countries})
dmutant

label                                             mutant  \
0           1  I have only see three episodes of Hack, starri...   
1           1  I have only see three episodes of Hack, starri...   
2           1  I have only see three episodes of Hack, starri...   
3           1  I have only see three episodes of Hack, starri...   
4           1  I have only see three episodes of Hack, starri...   
...       ...                                                ...   
986747      1  Well, the movie did turn out a lot better than...   
986748      1  Well, the movie did turn out a lot better than...   
986749      1  Well, the movie did turn out a lot better than...   
986750      1  Well, the movie did turn out a lot better than...   
986751      1  Well, the movie did turn out a lot better than...   

                                                 template  \
0       I have only see three episodes of Hack, starri...   
1       I have only see three episodes of Hack, starri...   
2       I have only see three episodes of Hack, starri...   
3       I have only see three episodes of Hack, starri...   
4       I have only see three episodes of Hack, starri...   
...                                                   ...   
986747  Well, the movie did turn out a lot better than...   
986748  Well, the movie did turn out a lot better than...   
986749  Well, the movie did turn out a lot better than...   
986750  Well, the movie did turn out a lot better than...   
986751  Well, the movie did turn out a lot better than...   

                                                 original identifier  type  \
0       I have only see three episodes of Hack, starri...       Roen  name   
1       I have only see three episodes of Hack, starri...       Jeet  name   
2       I have only see three episodes of Hack, starri...      Hagen  name   
3       I have only see three episodes of Hack, starri...     Willow  name   
4       I have only see three episodes of Hack, starri...      Belal  name   
...                                                   ...        ...   ...   
986747  Well, the movie did turn out a lot better than...       Dódi  name   
986748  Well, the movie did turn out a lot better than...     Iboyka  name   
986749  Well, the movie did turn out a lot better than...     Zsófia  name   
986750  Well, the movie did turn out a lot better than...      Ágota  name   
986751  Well, the movie did turn out a lot better than...      Alexa  name   

        gender  country  
0         male       UK  
1         male       UK  
2         male       UK  
3         male       UK  
4         male       UK  
...        ...      ...  
986747  female  Hungary  
986748  female  Hungary  
986749  female  Hungary  
986750  female  Hungary  
986751  female  Hungary  

[986752 rows x 8 columns]

In [362]:
import os

dirname = "../data/gc_imdb/"
# dirname = "../data/imdb_using_eec_name/"

if not os.path.exists(dirname) :
    os.makedirs(dirname)

dmutant.to_csv(dirname + "test.csv", index=None, header=None, sep="\t")

dmutant_shuffled = dmutant.sample(frac=1, random_state=123).reset_index(drop=True)
dmutant_shuffled.to_csv(dirname + "train.csv", index=None, header=None, sep="\t")

In [310]:
dmutant

label                                             mutant  \
0           1  I have only see three episodes of Hack, starri...   
1           1  I have only see three episodes of Hack, starri...   
2           1  I have only see three episodes of Hack, starri...   
3           1  I have only see three episodes of Hack, starri...   
4           1  I have only see three episodes of Hack, starri...   
...       ...                                                ...   
140837      1  Well, the movie did turn out a lot better than...   
140838      1  Well, the movie did turn out a lot better than...   
140839      1  Well, the movie did turn out a lot better than...   
140840      1  Well, the movie did turn out a lot better than...   
140841      1  Well, the movie did turn out a lot better than...   

                                                 template  \
0       I have only see three episodes of Hack, starri...   
1       I have only see three episodes of Hack, starri...   
2       I have only see three episodes of Hack, starri...   
3       I have only see three episodes of Hack, starri...   
4       I have only see three episodes of Hack, starri...   
...                                                   ...   
140837  Well, the movie did turn out a lot better than...   
140838  Well, the movie did turn out a lot better than...   
140839  Well, the movie did turn out a lot better than...   
140840  Well, the movie did turn out a lot better than...   
140841  Well, the movie did turn out a lot better than...   

                                                 original identifier  type  \
0       I have only see three episodes of Hack, starri...     Alonzo  name   
1       I have only see three episodes of Hack, starri...       Adam  name   
2       I have only see three episodes of Hack, starri...   Alphonse  name   
3       I have only see three episodes of Hack, starri...       Alan  name   
4       I have only see three episodes of Hack, starri...    Darnell  name   
...                                                   ...        ...   ...   
140837  Well, the movie did turn out a lot better than...    Melanie  name   
140838  Well, the movie did turn out a lot better than...    Tanisha  name   
140839  Well, the movie did turn out a lot better than...      Nancy  name   
140840  Well, the movie did turn out a lot better than...        Tia  name   
140841  Well, the movie did turn out a lot better than...  Stephanie  name   

        gender            country  
0         male   African American  
1         male   African American  
2         male   African American  
3         male   African American  
4         male   African American  
...        ...                ...  
140837  female  European American  
140838  female  European American  
140839  female  European American  
140840  female  European American  
140841  female  European American  

[140842 rows x 8 columns]

In [2]:
import pandas as pd

# dirname = "../data/gc_imdb/"
dirname = "../data/imdb_using_eec_name/"
dmutant = pd.read_csv(dirname + "test.csv", sep="\t", names=["label", "mutant", "template", "original", "identifier", "type", "gender", "country"])

In [3]:
dmutant

label                                             mutant  \
0           1  I have only see three episodes of Hack, starri...   
1           1  I have only see three episodes of Hack, starri...   
2           1  I have only see three episodes of Hack, starri...   
3           1  I have only see three episodes of Hack, starri...   
4           1  I have only see three episodes of Hack, starri...   
...       ...                                                ...   
140837      1  Well, the movie did turn out a lot better than...   
140838      1  Well, the movie did turn out a lot better than...   
140839      1  Well, the movie did turn out a lot better than...   
140840      1  Well, the movie did turn out a lot better than...   
140841      1  Well, the movie did turn out a lot better than...   

                                                 template  \
0       I have only see three episodes of Hack, starri...   
1       I have only see three episodes of Hack, starri...   
2       I have only see three episodes of Hack, starri...   
3       I have only see three episodes of Hack, starri...   
4       I have only see three episodes of Hack, starri...   
...                                                   ...   
140837  Well, the movie did turn out a lot better than...   
140838  Well, the movie did turn out a lot better than...   
140839  Well, the movie did turn out a lot better than...   
140840  Well, the movie did turn out a lot better than...   
140841  Well, the movie did turn out a lot better than...   

                                                 original identifier  type  \
0       I have only see three episodes of Hack, starri...     Alonzo  name   
1       I have only see three episodes of Hack, starri...       Adam  name   
2       I have only see three episodes of Hack, starri...   Alphonse  name   
3       I have only see three episodes of Hack, starri...       Alan  name   
4       I have only see three episodes of Hack, starri...    Darnell  name   
...                                                   ...        ...   ...   
140837  Well, the movie did turn out a lot better than...    Melanie  name   
140838  Well, the movie did turn out a lot better than...    Tanisha  name   
140839  Well, the movie did turn out a lot better than...      Nancy  name   
140840  Well, the movie did turn out a lot better than...        Tia  name   
140841  Well, the movie did turn out a lot better than...  Stephanie  name   

        gender            country  
0         male   African American  
1         male   African American  
2         male   African American  
3         male   African American  
4         male   African American  
...        ...                ...  
140837  female  European American  
140838  female  European American  
140839  female  European American  
140840  female  European American  
140841  female  European American  

[140842 rows x 8 columns]

In [4]:
dmutant["template"] = dmutant["template"].astype("category")
dmutant["template_id"] = dmutant["template"].cat.codes

In [5]:
dmutant

label                                             mutant  \
0           1  I have only see three episodes of Hack, starri...   
1           1  I have only see three episodes of Hack, starri...   
2           1  I have only see three episodes of Hack, starri...   
3           1  I have only see three episodes of Hack, starri...   
4           1  I have only see three episodes of Hack, starri...   
...       ...                                                ...   
140837      1  Well, the movie did turn out a lot better than...   
140838      1  Well, the movie did turn out a lot better than...   
140839      1  Well, the movie did turn out a lot better than...   
140840      1  Well, the movie did turn out a lot better than...   
140841      1  Well, the movie did turn out a lot better than...   

                                                 template  \
0       I have only see three episodes of Hack, starri...   
1       I have only see three episodes of Hack, starri...   
2       I have only see three episodes of Hack, starri...   
3       I have only see three episodes of Hack, starri...   
4       I have only see three episodes of Hack, starri...   
...                                                   ...   
140837  Well, the movie did turn out a lot better than...   
140838  Well, the movie did turn out a lot better than...   
140839  Well, the movie did turn out a lot better than...   
140840  Well, the movie did turn out a lot better than...   
140841  Well, the movie did turn out a lot better than...   

                                                 original identifier  type  \
0       I have only see three episodes of Hack, starri...     Alonzo  name   
1       I have only see three episodes of Hack, starri...       Adam  name   
2       I have only see three episodes of Hack, starri...   Alphonse  name   
3       I have only see three episodes of Hack, starri...       Alan  name   
4       I have only see three episodes of Hack, starri...    Darnell  name   
...                                                   ...        ...   ...   
140837  Well, the movie did turn out a lot better than...    Melanie  name   
140838  Well, the movie did turn out a lot better than...    Tanisha  name   
140839  Well, the movie did turn out a lot better than...      Nancy  name   
140840  Well, the movie did turn out a lot better than...        Tia  name   
140841  Well, the movie did turn out a lot better than...  Stephanie  name   

        gender            country  template_id  
0         male   African American         1218  
1         male   African American         1218  
2         male   African American         1218  
3         male   African American         1218  
4         male   African American         1218  
...        ...                ...          ...  
140837  female  European American         3417  
140838  female  European American         3417  
140839  female  European American         3417  
140840  female  European American         3417  
140841  female  European American         3417  

[140842 rows x 9 columns]

In [6]:
dmutant.groupby("template_id").count()

label  mutant  template  original  identifier  type  gender  \
template_id                                                                
0               40      40        40        40          40    40      40   
1               26      26        26        26          26    26      26   
2               40      40        40        40          40    40      40   
3               40      40        40        40          40    40      40   
4               40      40        40        40          40    40      40   
...            ...     ...       ...       ...         ...   ...     ...   
3725            40      40        40        40          40    40      40   
3726            40      40        40        40          40    40      40   
3727            40      40        40        40          40    40      40   
3728            26      26        26        26          26    26      26   
3729            40      40        40        40          40    40      40   

             country  
template_id           
0                 40  
1                  0  
2                 40  
3                 40  
4                 40  
...              ...  
3725              40  
3726              40  
3727              40  
3728               0  
3729              40  

[3730 rows x 8 columns]

In [7]:
dmutant.groupby("type").count()

label  mutant  template  original  identifier  gender  \
type                                                                          
gaw                   15158   15158     15158     15158       15158   15158   
name                 121880  121880    121880    121880      121880  121880   
person-pronoun-gaw      364     364       364       364         364     364   
person-pronoun-name     920     920       920       920         920     920   
person-pronoun-sltn      40      40        40        40          40      40   
sltn                   2480    2480      2480      2480        2480    2480   

                     country  template_id  
type                                       
gaw                        0        15158  
name                  121880       121880  
person-pronoun-gaw         0          364  
person-pronoun-name      920          920  
person-pronoun-sltn       40           40  
sltn                    2480         2480

In [8]:
NAME = "name"
SALUTATION = "sltn"
GAW = "gaw"
P = "person-pronoun-"
PNAME = P + NAME
PGAW = P + GAW
PSALUTATION = P + SALUTATION

In [43]:
name = dmutant.groupby("type").get_group(NAME)

In [44]:
name = name.iloc[:,[2,3]]

In [46]:
name = name.drop_duplicates().reset_index(drop=True)

In [139]:
id = 49
id = 62
id = 72
id = 78
id = 81
id = 83
name["template"][id]

"According to me, a movie can be best rated by the number of ' best scenes ' there are. There are probably at least 8090 scenes in the whole movie that is probably rated as ' best ' by somebody or the other. This movie is a masterpiece. Period. I know many who have called it ahead of its time, probably because it bombed at boxoffice. However, I think any generation can relate to this movie and have a good laugh through out. I am personally glad that <name> is part of the movie too. <pro-His> role isn't that ' meaty ', however its good to see that <pro-he> is a part of the cast in the funniest Hindi movie ever."

In [140]:
name["original"][id]

"According to me, a movie can be best rated by the number of ' best scenes ' there are. There are probably at least 8090 scenes in the whole movie that is probably rated as ' best ' by somebody or the other. This movie is a masterpiece. Period. I know many who have called it ahead of its time, probably because it bombed at boxoffice. However, I think any generation can relate to this movie and have a good laugh through out. I am personally glad that Mahmood is part of the movie too. His role isn't that ' meaty ', however its good to see that he is a part of the cast in the funniest Hindi movie ever."

In [39]:
id = 0
print(dmutant.groupby("type").get_group(NAME).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(NAME).iloc[id]["template"])

I have only see three episodes of Hack, starring David Morse, and it looks as though I've missed 37 episodes! well that s if ITV 3 are showing them in chronological order. I've just watched ' Misty Blue ' ( episode 38). I have really enjoyed the 3 episodes, but then I'm a David Morse fan, ( esp St. Elsewhere). For any one reading this, Hack is excellent. Pity its being aired on ITV 3. The cast is strong, though I ca nt get used to the idea of David playing a bent cop, still we all know he's the good guy wrongly accused. I see Gary Cole has guest starred, what ever happened to ' Midnight Caller '? Just wonder if Hack is available on DVD ( yet). Lovin it. Cheers.

I have only see three episodes of Hack, starring David Morse, and it looks as though I've missed 37 episodes! well that s if ITV 3 are showing them in chronological order. I've just watched ' Misty Blue ' ( episode 38). I have really enjoyed the 3 episodes, but then I'm a David Morse fan, ( esp St. Elsewhere). For any one readi

In [37]:
id = 520
print(dmutant.groupby("type").get_group(NAME).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(NAME).iloc[id]["template"])

Just when you think that you've seen the worst the zombie subgenre can offer, along comes another budding Romero and his team of notalent mates to prove you wrong. I've suffered the Zombie Bloodbath trilogy, endured Death Valley: The Revenge of Bloody Bill, sat semicomatose through The Zombie Diaries, and laughed hysterically during Zombie ' 90: Extreme Pestilence, and genuinely thought that amateur moviemaking could sink no lower. However, having just watched Zombie Planet, a two hour long festering pile of drivel from writer/ director George Bonilla, I think I have found the ultimate in awful zombie movies. In this dreadfully amateurish effort, which is part Mad Max and part Dawn of the Dead ( but ALL bad), Frank Farhat stars as T. K. Kane, a tough fighter who kicks zombie ass for a hobby, armed only with knives, a machete, some concussion grenades, and what he obviously perceives to be a really mean glare. After a desperately unexciting opening sequence in which he takes on a horde 

In [14]:
id = 9000
print(dmutant.groupby("type").get_group(NAME).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(NAME).iloc[id]["template"])

How truly sad that this sprung from the same mind as Donnie Darko, possibly one of the best films in this genre. Where do I even begin? I think one of the must infuriating aspects of the film is that we are supposed to be critiquing humanity, and yet we see no humanity in the film. No more than 5 minutes of the film is spend agonizing about the possible death of another human. These are horrible one dimensional cardboard cutouts of human beings. Sadly, that's how they are played with what can only be described as dreadful acting. Is this truly how Kelly sees humanity? Judging from the reaction of viewers, this is a horrible encapsulation of humanity. Why don't the characters in the film ask the questions that all the viewers have? This is not an indictment of humanity. It's an indictment of the straw men that Kelly sets up who bare almost no resemblance to real humans in this situation. To those who say this was a wonderful thoughtprovoking film, to what are you comparing it? Armageddo

In [173]:
dmutant.groupby("type").get_group(SALUTATION)

label                                             mutant  \
29324       0  The James Dean Story is introduced as " A diff...   
29325       0  The James Dean Story is introduced as " A diff...   
29326       0  The James Dean Story is introduced as " A diff...   
29327       0  The James Dean Story is introduced as " A diff...   
29328       0  The James Dean Story is introduced as " A diff...   
...       ...                                                ...   
963545      1  Although it isn't mentioned very often, " Don'...   
963546      1  Although it isn't mentioned very often, " Don'...   
963547      1  Although it isn't mentioned very often, " Don'...   
963548      1  Although it isn't mentioned very often, " Don'...   
963549      1  Although it isn't mentioned very often, " Don'...   

                                                 template  \
29324   The James Dean Story is introduced as " A diff...   
29325   The James Dean Story is introduced as " A diff...   
29326   The James Dean Story is introduced as " A diff...   
29327   The James Dean Story is introduced as " A diff...   
29328   The James Dean Story is introduced as " A diff...   
...                                                   ...   
963545  Although it isn't mentioned very often, " Don'...   
963546  Although it isn't mentioned very often, " Don'...   
963547  Although it isn't mentioned very often, " Don'...   
963548  Although it isn't mentioned very often, " Don'...   
963549  Although it isn't mentioned very often, " Don'...   

                                                 original identifier  type  \
29324   The James Dean Story is introduced as " A diff...       Roen  sltn   
29325   The James Dean Story is introduced as " A diff...       Jeet  sltn   
29326   The James Dean Story is introduced as " A diff...      Hagen  sltn   
29327   The James Dean Story is introduced as " A diff...     Willow  sltn   
29328   The James Dean Story is introduced as " A diff...      Belal  sltn   
...                                                   ...        ...   ...   
963545  Although it isn't mentioned very often, " Don'...       Dódi  sltn   
963546  Although it isn't mentioned very often, " Don'...     Iboyka  sltn   
963547  Although it isn't mentioned very often, " Don'...     Zsófia  sltn   
963548  Although it isn't mentioned very often, " Don'...      Ágota  sltn   
963549  Although it isn't mentioned very often, " Don'...      Alexa  sltn   

        gender  country  template_id  
29324     male       UK         2526  
29325     male       UK         2526  
29326     male       UK         2526  
29327     male       UK         2526  
29328     male       UK         2526  
...        ...      ...          ...  
963545  female  Hungary          234  
963546  female  Hungary          234  
963547  female  Hungary          234  
963548  female  Hungary          234  
963549  female  Hungary          234  

[19220 rows x 9 columns]

In [174]:
id = 0
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

The James Dean Story is introduced as " A different kind of motion picture, " explaining, " The presence of the leading character in this film has been made possible by the use of existing motion picture material, tape recordings of his voice and by means of a new techniquedynamic exploration of the still photograph. " The only " tape recordings of his voice " noteworthy is one short recording Mr. Dean make while visiting his family in Indiana; he wanted to record any family recollections of his greatgrandfather Cal Dean, intrigued because he played a similarly named " Cal " in " East of Eden ". Dean asks if Cal Dean was interested in art, and learns the relative was an auctioneer. James Dean was interested in art and had warm relationship with his family, obviously. That's the only 100% accurate revelation in this documentary. James Dean was interested in art and had warm relationship with his family. An amazing " screen test"/"outtake " from " East of Eden " appears near the film's e

In [175]:
id = 400
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

This is one of my favorite T.V shows of all time, Rowan Atkinson is simply a genius!, and it's only fitting that i chose this to be my 1000 review!. I can't begin to tell you how much i love Mr. Bean he's the man, and what amazes me, is how he gets out of these incredibly difficult situations, and he is always so creative, plus Robyn Driscoll also deserves accolades!, he is also a genius!. My favorite bit that he has done is the Amazing Adventures of Mr. Bean and while all the rest of them are amazing, this remains my true favorite, plus i wish the show didn't stop so soon!. Each episode is brilliantly written, and they were all masterfully directed, plus Each episode is a classic in my eyes!. This show is incredibly popular, and i can definitely see why, as it's quite possibly the funniest show ever. The character actors all played there roles really well, especially Robyn Driscoll and Matilda Ziegler ( as Irma). This is one of my favorite T.V shows of all time Rowan Atkinso is simply

In [178]:
id = 1200
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

The New Twenty is one of the worst films I have ever seen. Yes, some may argue that formulaic small budget films that strive for less may seem worse, but I would argue that a pseudointellectual antiformulaic " indie " film that pretends to be more is worse.   " The New Twenty " was written and directed by Chris Mason Johnson, and I will never get back the 91 minutes of my life that I wasted on his film. THE SCREENPLAY & CHARACTERS From beginning to end, the screenplay failed to provide an antihero or hero with a moral core that the audience would want to see succeed or fail.   In story telling, the three dramatic conflicts are man vs. man, man vs. the world, and man vs. himself.   The screenplay focuses on a group of college friends in New York, and is preoccupied with the dynamic of man vs. himself, where each of the film's characters are so selfabsorbed in a stupor of depression and selfdestructive behavior that the movie atrophies before the audiences very eyes.   Each of the main c

In [179]:
id = 1600
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

I have read almost all the books by now, and have seen the musical production in two different languages. I absolutely adored everything that I have been acquainted previously. But lately I've been running out of resources to sustain my fancy. I still have couple of books left, but they are either in transit or they are the sequels which I am not in a hurry to read. So an idea dawned on meSink me! There are movies which I have not watched! Thus, I must watch them immediately. The first five minutes of these series were ... acceptable. In fact, I quite enjoyed the variation of the reason for the denunciation, however different it had seemed. It went all downhill from there, though. Chauvelin was too ... foppish? And, I daresay, too old for this role. Not nearly that dark and dashing figure with his dreadful eitheror. ( And what was that with random bed scene featuring him? It was way too creepywasn't his only love the Madame Guillotine, and his only interesthis job, and his only obsessi

In [180]:
id = 2000
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

It's hard to know exactly what to say about this ever so bland and dull little film. The story is predictable when not completely laughable. It's all a matter of " dutiful gestures " which, as presented here, carry absolutely no conviction. Yes, the MGM " production values " are gorgeous, and yes, Ms. Lamarr was exquisitely beautiful, but she and the great Spencer Tracy have absolutely no " chemistry " togetherand that's the only thing that would have made this parade of cliches at all effective ... It's my understanding that this movie received poor reviews when it was originally released; the passage of time has not improved it.

It's hard to know exactly what to say about this ever so bland and dull little film. The story is predictable when not completely laughable. It's <sltn-Ms.> <name> . Yes, the MGM " production values " are gorgeous, and yes, <sltn-Ms.> <name> was exquisitely beautiful, but <pro-she> and the great Spencer Tracy have absolutely no " chemistry " togetherand that

In [181]:
id = 2500
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

Mr. Bean has shaped the face of British TV comedy. He has proved that you do not need wicked words or wit, a massive budget, a great deal of intelligence or even any intelligence to make something brilliant. And Mr. Bean is one of those characters who you just can't forget. Some of these episodes had me in stitchesyes, they're not realistic at all and they're all pretty stupid, but to be honest, realism is one of the barriers Bean has broken on its way to greatness. Rowan Atkinson and co. always manage to cook up interesting new ideasand hilarious new gagsremember when Mr. Bean drove his green Mini whilst sitting on a sofa on the roof? Mr. Bean is one of those things that never gets weakthe movie wasn't as good as this, but Bean has introduced a distinct new sense of humour to the world, and kids and adults alike will marvel at its immense fun factor. " Extras " and " Little Britain " can be damnedthis is British comedy at its best and most original. These escapades never get old! 10/1

In [177]:
id = 4000
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

Mr. Moto's Gamble has a fairly straight forward plotwhen a boxer is murdered in the ring with a mysterious poison, it's up to the even more mysterious Mr. Moto to solve the case. I'm shocked at the number of positive reviews for Mr. Moto's Gamble on IMDb. Because to me ... well, I found it extremely disappointing. I enjoy Mr. Moto and I enjoy Charlie Chan, but I can't say I cared for this mishmash of the two. For those unfamiliar with the story behind Mr. Moto's Gamble, it was originally intended to be a Charlie Chan film. But when Warner Oland backedout, some of the scenes and action were rewritten for Peter Lorre and Mr. Moto. As I indicated, the end result left me underwhelmed. Mr. Moto is not Chan. He's more mysterious, he's more athletic, and he's more exotic. So trying to put Moto in a Chan film is like trying to fit a square peg into a round holeit doesn't work. And listening to Lorre/ Moto try to deliver one of Chan's trademark euphemisms just ends up sounding silly. Add to tha

In [ ]:
id = 4000
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(SALUTATION).iloc[id]["template"])

In [117]:
dmutant.groupby("type").get_group(GAW)

label                                             mutant  \
48       0  I'm sorry but I can't agree that this was a go...   
49       0  I'm sorry but I can't agree that this was a go...   
50       0  I'm sorry but I can't agree that this was a go...   
51       0  I'm sorry but I can't agree that this was a go...   
52       0  I'm sorry but I can't agree that this was a go...   
..     ...                                                ...   
793      1  L'Auberge Espagnole collected the audience whe...   
794      1  L'Auberge Espagnole collected the audience whe...   
795      1  L'Auberge Espagnole collected the audience whe...   
796      1  L'Auberge Espagnole collected the audience whe...   
797      1  L'Auberge Espagnole collected the audience whe...   

                                              template  \
48   I'm sorry but I can't agree that this was a go...   
49   I'm sorry but I can't agree that this was a go...   
50   I'm sorry but I can't agree that this was a go...   
51   I'm sorry but I can't agree that this was a go...   
52   I'm sorry but I can't agree that this was a go...   
..                                                 ...   
793  L'Auberge Espagnole collected the audience whe...   
794  L'Auberge Espagnole collected the audience whe...   
795  L'Auberge Espagnole collected the audience whe...   
796  L'Auberge Espagnole collected the audience whe...   
797  L'Auberge Espagnole collected the audience whe...   

                                              original identifier type  \
48   I'm sorry but I can't agree that this was a go...        boy  gaw   
49   I'm sorry but I can't agree that this was a go...    brother  gaw   
50   I'm sorry but I can't agree that this was a go...      daddy  gaw   
51   I'm sorry but I can't agree that this was a go...        man  gaw   
52   I'm sorry but I can't agree that this was a go...     father  gaw   
..                                                 ...        ...  ...   
793  L'Auberge Espagnole collected the audience whe...       aunt  gaw   
794  L'Auberge Espagnole collected the audience whe...       girl  gaw   
795  L'Auberge Espagnole collected the audience whe...        mom  gaw   
796  L'Auberge Espagnole collected the audience whe...       lady  gaw   
797  L'Auberge Espagnole collected the audience whe...       lady  gaw   

     gender country  template_id  
48     male    None           39  
49     male    None           39  
50     male    None           39  
51     male    None           39  
52     male    None           39  
..      ...     ...          ...  
793  female    None           48  
794  female    None           48  
795  female    None           48  
796  female    None           48  
797  female    None           48  

[494 rows x 9 columns]

In [118]:
id = 0
print(dmutant.groupby("type").get_group(GAW).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(GAW).iloc[id]["template"])

I'm sorry but I can't agree that this was a good movie. Yes, it looked good visually, but it's the story that drives the movie and I must say the story sucked bigtime. How in the world did they manage to slip some of those plotholes past the critics. Better story and I would've gave it a higher vote but I was impossible to do that and still be able to live with myself. I have always been a fan of scary movies, and the previews really had me fooled. All the scary scenes were shown in the previews. And why did the family that got killed stay to haunt the house? Why did the father come back again? WHy did he decide to kill in the first place? Why were the kids the only ones to see the ghosts first? To many questions, not enough answers. If I could've gave it a zero, I would've.

I'm sorry but I can't agree that this was a good movie. Yes, it looked good visually, but it's the story that drives the movie and I must say the story sucked bigtime. How in the world did they manage to slip some

In [123]:
id = 1000
print(dmutant.groupby("type").get_group(GAW).iloc[id]["original"])
print()
print(dmutant.groupby("type").get_group(GAW).iloc[id]["template"])

IndexError: single positional indexer is out-of-bounds